### PR TITLE
3480 worldmap links

### PR DESCRIPTION
**Pull request by pdurbin**
# RFI Checklist

### 1. Related Issues

- connects to #3480 Dataverse: script to verify WorldMap links are valid

## Pull Request Checklist

- [x] Unit [tests][] completed: None
- [x] Integration [tests][]: None
- [x] Deployment requirements, [SQL updates][], [Solr updates][], etc.: upgrade_v4.6.1_to_v4.6.2.sql
- [x] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/4.6.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/api/Files.java [hunk 1]
```diff
@@ -216,8 +217,37 @@ public Response replaceFileInDataset(
...
+    /**
+     * @todo Enforce permissions!
+     */
+    @GET
+    @Path("{id}/map")
+    public Response getMapFromFile(@PathParam("id") Long idSupplied) {
+        DataFile dataFile = fileService.find(idSupplied);
+        MapLayerMetadata mapLayerMetadata = mapLayerMetadataSrv.findMetadataByDatafile(dataFile);
+        return ok(json(mapLayerMetadata));
+    }
```
Comment (raprasad) [inline src/main/java/edu/harvard/iq/dataverse/api/Files.java]: Check for nulls

```java
DataFile dataFile = fileService.find(idSupplied);
if (dataFile == null){  
//  404 
}

MapLayerMetadata mapLayerMetadata = mapLayerMetadataSrv.findMetadataByDatafile(dataFile);
if (mapLayerMetadata == null){  
//  404 
}
```

Comment (raprasad) [inline src/main/java/edu/harvard/iq/dataverse/api/Files.java]: fyi: checked with @pdurbin who knows about this

File: src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java [hunk 2]
```diff
@@ -788,4 +788,26 @@ public static JsonObjectBuilder getChecksumTypeAndValue(DataFile.ChecksumType ch
...
+                .add("id", mapLayerMetadata.getId())
+                .add("embedMapLink", mapLayerMetadata.getEmbedMapLink())
+                .add("joinDescription", mapLayerMetadata.getJoinDescription())
+                .add("layerLink", mapLayerMetadata.getLayerLink())
+                .add("layerName", mapLayerMetadata.getLayerName())
+                .add("mapImageLink", mapLayerMetadata.getMapImageLink())
+                .add("mapLayerLinks", Json.createReader(new StringReader(mapLayerMetadata.getMapLayerLinks())).readObject())
+                .add("worldmapUsername", mapLayerMetadata.getWorldmapUsername())
+                .add("isJoinLayer", mapLayerMetadata.isJoinLayer())
+                .add("dataFileId", mapLayerMetadata.getDataFile().getId())
```
Comment (raprasad) [inline src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java]: Add link to the File page that uses the dataFileId:
```
// example url
https://dataverse.harvard.edu/file.xhtml?fileId={dataFileId}
```

e.g. ```filePageLink``` or current naming standard

```java

String filePageLink = "{domain/server}/file.xhtml?fileId=" + mapLayerMetadata.getDataFile().getId();

// within return jsonObjectBuilder()
             .add("filePageLink", filePageLink)
```

Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java]: good idea but I put it in what I feel is a more useful place: https://github.com/IQSS/dataverse/blob/2ca06dca33e62a5a37c6ef1339e2a1d0b65c8229/src/main/java/edu/harvard/iq/dataverse/api/Geoconnect.java#L48

**Comment by pdurbin** [comment]
@scolapasta thanks for the comments. The easiest way to move forward is to simply delete both the `GET /map` and `GET /map/check` endpoints from the Files API. (We would keep `DELETE /map` which uses a command.) Deleting those `GET`'s would mean there would be no need to write another command. I think virtually no one is going to use them anyway. I hope I explained well enough at https://github.com/IQSS/dataverse/blob/2ca06dca33e62a5a37c6ef1339e2a1d0b65c8229/doc/sphinx-guides/source/admin/geoconnect-worldmap.rst it is expected that a **sysadmin** will be doing the deletes rather than **end users**. I expect we'll only have a handful of deletes a year, if any, and a sysadmin will do them. I don't want to expand the scope of the issue and, again, deleting the `GET`'s above is probably the fastest way forward.

**Pull request merged**